### DISCUSSION TITLE
How can I use websocket in microservice?

### DISCUSSION THREAD

**Discussion by Marco24082001** [jhipster/generator-jhipster#28324] [Q&A]
I want to add realtime chat functionality, but I don't know how to implement socket in microservice, pls help me.

**Comment by TatyOko28** [comment]
In a microservice architecture, you usually should not put WebSocket logic inside every microservice.

A common approach is to create a dedicated **WebSocket Gateway / Realtime Service**.

The architecture can be:

```text
Frontend
   |
   | WebSocket connection
   v
Realtime Gateway Service
   |
   | REST / gRPC / Message Broker
   v
Chat Service / User Service / Notification Service
